vmss-prototype/vmss-prototype

```diff
@@ -352,6 +352,98 @@ def get_vmss_images(subscription, resource_group, sig_name, image_definition):
     return json.loads(output)
 
 
+def image_tweaks(node_name):
+    """
+    If we have the information needed, do our image tweaks pod on the
+    target node.  This runs as a one-shot pod that will address image
+    tweaks.  Currently, that is only 1 item - the /etc/machine-id must
+    be emptied (but not deleted) such that a new unique machine-id can
+    be created for each new node.
+    The yaml specifically supports running on a cordoned node and it
+    also specifically calls out the node in question.
+    :param node_name:  The node, after it has been drained but it is still up
+    :returns: True if the tweaks were successfully deployed
+    """
+    namespace = os.getenv('POD_NAMESPACE', None)
+    if not namespace:
+        logging.info('No POD_NAMESPACE defined, can not do image tweaks')
+        return False
+
+    image = os.getenv('POD_IMAGE', None)
+    if not image:
+        logging.info('No POD_IMAGE defined, can not do image tweaks')
+        return False
+
+    pod_name = 'vmss-prototype-tweaks-{0}'.format(node_name)
+
+    # This is a pod yaml for our image tweak we will do on the target node
+    # that will be our new prototype image.
+    yaml_text = ("apiVersion: v1\n"
+                 "kind: Pod\n"
+                 "metadata:\n"
+                 "  name: '{0}'\n"
+                 "  namespace: '{1}'\n"
+                 "spec:\n"
+                 "  containers:\n"
+                 "  - command:\n"
+                 "    - /bin/cp\n"
+                 "    - /dev/null\n"
+                 "    - /hostetc/machine-id\n"
+                 "    image: '{2}'\n"
+                 "    imagePullPolicy: IfNotPresent\n"
+                 "    name: tweaks\n"
+                 "    volumeMounts:\n"
+                 "    - mountPath: /hostetc\n"
+                 "      name: etc\n"
+                 "      readOnly: false\n"
+                 "  hostNetwork: true\n"
+                 "  imagePullSecrets:\n"
+                 "  - name: skyman-acr\n"
+                 "  nodeSelector:\n"
+                 "    kubernetes.io/hostname: '{3}'\n"
+                 "  priorityClassName: skyman-cluster-daemonset\n"
```

Comment:
does this exist?

Reply:
Oops, in my clusters :-)
Will remove that.